### ISSUE TITLE
Crash after selecting origin 1.18.2

### ISSUE THREAD

**Issue by willko22** [EdwinMindcraft/origins-architectury#222]
The game crashed whilst ticking player
Error: java.lang.ClassCastException: class net.minecraft.server.level.ServerPlayer cannot be cast to class io.github.apace100.apoli.mixin.EntityAccessor (net.minecraft.server.level.ServerPlayer is in module minecraft@1.18.2 of loader 'TRANSFORMER' @40fa8766; io.github.apace100.apoli.mixin.EntityAccessor is in module apoli@1.18.2-2.3.4.1 of loader 'TRANSFORMER' @40fa8766)
Exit Code: -1

**Comment by EdwinMindcraft** [comment]
You know what, fine. I'll just archive 1.5.0.2.

**Comment by EdwinMindcraft** [comment]
Downgrade to 1.5.0.1.

**Comment by willko22** [comment]
Haha. Too many problems for release right?
It's ok. If you are u correct them or a that version then it will be nice.

**Comment by EdwinMindcraft** [comment]
JiJ breaks mixins and servers, I need to PR forge to fix it, and I don't exactly want to do that currently.

**Comment by willko22** [comment]
It's ok. Everyone has their circumstances when they don't or do want to do something. If 1.5.0.1 works then there is no need to correct 1.5.0.2 immediately.

I have noticed that no matter what I do I have broken mixin on the server.

What is JiJ?

Idk if it's ok to ask this, why do origin classes need apotheosis?

**Comment by EdwinMindcraft** [comment]
JiJ is forge's system to allow a mod jar to have other jars contained inside it, the main issue is that 1.5.0.1 does not work properly, since `apoli:adjacent` is broken (again, I've stopped counting how many times I've broken that power), which means felines are broken.

The forge fix is rather simple, but I can't be bothered to spend 30 minutes decompiling minecraft again just to change five lines of code. (It's just me being lazy and 1.18.2 not being the active version of minecraft).

For the server, I don't know which mixin causes issues, but Origins (apoli) has some mixins that are disabled by default, or depend on a given version of a mod, so that may be the issue.

As for why classes requires Apotheosis, you should ask LimonBlaze, but it may just be an oversight on their end.
Basically Apotheosis changes how enchanting works quite significantly, which would break the enchanter (I think that's what it's called) by preventing it's powers from working. Since Apotheosis is a widely used mod (basically it contains a lot of QoL changes and is a staple of most 1.12+ modpacks) they probably deemed it necessary to support the mod.
Given that classes is compiled and ran against Apotheosis in dev (as far as I can tell), it is possible that some code that should've been unreachable if the mod was not found is currently reachable and has slipped under the radar.

**Comment by RaymondBlaze** [comment]
> Idk if it's ok to ask this, why do origin classes need apotheosis?

EdwinMindcraft has basically explained the whole thing, but let me make some further explanation:
- Origins: Classes (Forge) doesn't need Apotheosis at runtime, the CurseForge page has only marked it as an **Optional** Dependency. This is basically to let players know Origins: Classes (Forge) has active support and code interaction for that mod.
- I'm quite open to work on needed mod compatibilities: [Origins: Classes (Forge) 1.18.2-1.1.6](https://github.com/LimonBlaze/origins-classes-forge/releases/tag/1.1.6) has compatibilities with Apotheosis (for the Cleric class), Tetra (for the Blacksmith class), Farmer's Delight (for the Cook class),  AppleSkin (for the Cook class and all ModifyFoodPower).
- If you feel like there's some mod compatibilities that must be solved by the mod itself rather than configs or datapacks, feel free to open an issue or pr.

**Comment by willko22** [comment]
thank you for the answer.

so 20. 8. 2022 o 15:39 LimonBlaze ***@***.***> napísal(a):

**Comment by willko22** [comment]
I have noticed that no matter what I do I have broken mixin on the server.

What is JiJ?

Idk if it's ok to ask this, why do origin classes need apotheosis?
